agenda/presentation/src/main/java/com/gkp/agenda/presentation/task/edittask/EditTaskScreen.kt

```diff
@@ -128,9 +204,167 @@ private fun EditTaskScreen(
 
     }
 
+    TaskyDatePickerDialog(
+        showDatePicker = showDatePicker,
+        state = datePickerState,
+        dismissDatePicker = { showDatePicker = false },
+        onDateSelected = onTaskDateSelected
+
+    )
+    TaskyTimePickerWithDialog(
+        state = timePickerState,
+        showTimePicker = showTimePicker,
+        onTimeSelected = onTaskTimeSelected,
+        dismissTimePicker = { showTimePicker = false }
+    )
 }
 
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun TaskyDatePickerDialog(
+    showDatePicker: Boolean,
+    dismissDatePicker: () -> Unit,
+    onDateSelected: (Long) -> Unit,
+    state: DatePickerState,
+) {
+    if (showDatePicker) {
+        DatePickerDialog(
+            onDismissRequest = {
+                dismissDatePicker()
+            },
+            confirmButton = {
+                TextButton(
+                    onClick = {
+                        dismissDatePicker()
+                        state.selectedDateMillis?.let {
+                            onDateSelected(it)
+                        }
+                    }
+                ) {
+                    Text(text = stringResource(R.string.date_picker_ok))
+                }
+            },
+        ) {
+            DatePicker(
+                state = state,
+            )
+        }
+    }
+}
 
+@Composable
+fun ReminderDropDownMenu(
+    modifier: Modifier = Modifier,
+    expanded: Boolean,
+    onDismissRequest: () -> Unit,
+    onCLickReminderMenuItem: (Int) -> Unit,
+) {
+    DropdownMenu(
+        modifier = modifier,
+        expanded = expanded,
+        onDismissRequest = { onDismissRequest() }
+    ) {
+        reminderTimeList.forEachIndexed { index, reminderTimeText ->
+            DropdownMenuItem(
+                text = {
+                    Text(
+                        text = reminderTimeText,
+                        style = MaterialTheme.typography.bodyLarge.copy(
+                            fontWeight = FontWeight.W400
+                        )
+
+                    )
+                },
+                onClick = {
+                    onCLickReminderMenuItem(index)
+                    onDismissRequest()
+                }
+            )
+        }
+
+    }
+}
+
+val reminderTimeList = listOf(
+    "10 minutes before",
+    "30 minutes before",
+    "1 hour before",
+    "6 hour before",
+    "1 day before"
+)
```

Comment:
Could you structure this differently so that you could still support localization? 🤔